src/common/users.ts

```diff
@@ -339,3 +350,51 @@ export const getUserReadingRank = async (
     tags,
   };
 };
+
+export const getStreakLastView = async (
+  con: DataSource | EntityManager,
+  streak: GQLUserStreak,
+): Promise<Date> => {
+  if (streak.lastViewAt) {
+    return streak.lastViewAt;
+  }
+
+  const view = await con
+    .getRepository(View)
+    .findOneBy({ userId: streak.userId });
+
+  return view?.timestamp;
+};
+
+export enum Day {
+  Sunday,
+  Monday,
+  Tuesday,
+  Wednesday,
+  Thursday,
+  Friday,
+  Saturday,
+}
+
+export const WEEKENDS = [Day.Sunday, Day.Saturday];
+export const FREEZE_DAYS_IN_A_WEEK = WEEKENDS.length;
+
+export const clearThenGetUserStreak = (
+  ctx: Context,
+  info: GraphQLResolveInfo,
+  userId: string,
+) => {
+  return ctx.con.transaction(async (manager) => {
+    await manager
+      .getRepository(UserStreak)
+      .update({ userId }, { currentStreak: 0 });
```

Comment:
We should also update the `updatedAt` column with the current date.

Reply:
This should have happened automatically. I instead introduced a decorator for when the entity gets updated, this column should get updated too.

https://github.com/dailydotdev/daily-api/pull/1658/commits/05fc9c296358b0875c38536a4b9804a1f313bc88

Reply:
> This should have happened automatically. I instead introduced a decorator for when the entity gets updated, this column should get updated too.

@sshanzel I don't think this applies on such updates. Can you please confirm?

Reply:
Found some issue related to this https://github.com/typeorm/typeorm/issues/5385

It is probably not best to rely on the decorator.